File: Julia/code_0406.jl
function fibo(N)
    println("fibo($N)を呼び出しました．")

    # ベースケース
    if N == 0
        return 0
    elseif N == 1
        return 1
    else
        result = fibo(N - 1) + fibo(N - 2);
        println("fibo($N) = $result")

        """
        println()で処理を終えると，関数がNothing型になるので，@ showやreturnで数値を返すこと．
        """

        return result
    end
end


fibo(6)

File: C++/code_0406.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <stdio.h>
using namespace std;

int fibo(int N)
{
    //再帰関数を呼び出したことを報告する
    cout << "fibo(" << N << ")を呼び出しました．" << endl;

    //ベースケース
    if (N == 0)
    {
        return 0;
    }
    else if (N == 1)
    {
        return 1;
    }
    else
    {
        int result = fibo(N - 1) + fibo(N - 2);
        cout << "fibo(" << N << ") = " << result << endl;
        return result;
    }
}

int main()
{
    fibo(20);
}